contracts/modules/maker/MakerV2Loan.sol

```diff
@@ -0,0 +1,626 @@
+pragma solidity ^0.5.4;
+
+import "./MakerV2Base.sol";
+import "../../defi/Loan.sol";
+import "../../defi/utils/MakerRegistry.sol";
+
+contract JugLike {
+    function drip(bytes32) external;
+}
+
+interface IUniswapFactory {
+    function getExchange(address _token) external view returns(IUniswapExchange);
+}
+
+interface IUniswapExchange {
+    function getEthToTokenOutputPrice(uint256 _tokens_bought) external view returns (uint256);
+    function getEthToTokenInputPrice(uint256 _eth_sold) external view returns (uint256);
+    function getTokenToEthOutputPrice(uint256 _eth_bought) external view returns (uint256);
+    function getTokenToEthInputPrice(uint256 _tokens_sold) external view returns (uint256);
+}
+
+/**
+ * @title MakerV2Loan
+ * @dev Module to migrate old CDPs and open and manage new CDPs.
+ * @author Olivier VDB - <olivier@argent.xyz>
+ */
+contract MakerV2Loan is Loan, MakerV2Base {
+
+    // The address of the MKR token
+    GemLike internal mkrToken;
+    // The address of the WETH token
+    GemLike internal wethToken;
+    // The address of the WETH Adapter
+    JoinLike internal wethJoin;
+    // The address of the Jug
+    JugLike internal jug;
+    // The address of the CDP Manager
+    ManagerLike internal cdpManager;
+    // The address of the SCD Tub
+    SaiTubLike internal tub;
+    // The Maker Registry in which all supported collateral tokens and their adapters are stored
+    MakerRegistry internal makerRegistry;
+    // The Uniswap Exchange contract for DAI
+    IUniswapExchange internal daiUniswap;
```

Comment:
We know that Uniswap is working on a V2 that should be released around february or march. We need to anticipate that change.

Reply:
For the record we have decided to leave it as it is. Users will be able to use the atomic swap of ETH/MKR for CDP migration and ETH/DAI to repay their debt for as long as there will be liquidity on Uniswap. We anticipate that:
1. all our CDP users will have migrated their CDP by the time the ETH/MKR liquidity is too low on Uniswap.
2. the client will handle the ETH/DAI swap before repaying debt when the ETH/DAI Uniswap pool is empty.